testsuite/modern-testsuite/src/test/java/org/sonatype/nexus/testsuite/npm/rebuild/RebuildNpmMetadataIT.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Sonatype Nexus (TM) Open Source Version
+ * Copyright (c) 2008-2015 Sonatype, Inc.
+ * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
+ * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.nexus.testsuite.npm.rebuild;
+
+import java.io.File;
+
+import org.sonatype.nexus.client.core.subsystem.content.Location;
+import org.sonatype.nexus.testsuite.client.Scheduler;
+import org.sonatype.nexus.testsuite.npm.NpmITSupport;
+
+import com.bolyuba.nexus.plugin.npm.client.NpmHostedRepository;
+import com.google.common.base.Charsets;
+import com.google.common.io.Files;
+import org.apache.commons.io.FileUtils;
+import org.junit.Test;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.sonatype.sisu.litmus.testsupport.hamcrest.FileMatchers.exists;
+
+/**
+ * Rebuild NPM metadata.
+ */
+public class RebuildNpmMetadataIT
+    extends NpmITSupport
+{
+  public RebuildNpmMetadataIT(String nexusBundleCoordinates) {
+    super(nexusBundleCoordinates);
+  }
+
+  @Test
+  public void rebuildNpmMetadata() throws Exception {
+    final NpmHostedRepository hostedRepository = createNpmHostedRepository(testMethodName());
+    final File hostRepositoryStorage = new File(nexus().getWorkDirectory(), "storage/" + hostedRepository.id());
+    FileUtils.copyFile(testData().resolveFile("commonjs-0.0.1.tgz"),
+        new File(hostRepositoryStorage, "commonjs/-/commonjs-0.0.1.tgz"), false);
```

Comment:
Yes, that's part of npm repo spec, at least host NX hosted repository implements it.